### PR TITLE
Implemented oms3_deleteConnectorFromBus

### PR DESCRIPTION
**Pull request by adeas31**
### Purpose

Delete the connectors from the bus.

### Approach

Implemented `oms3_deleteConnectorFromBus` API.

### Type of Change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

### Checklist

- I have performed a self-review of my own code

### CODE REVIEW

File: src/OMSimulatorLib/BusConnector.cpp [hunk 1]
```diff
@@ -100,6 +100,20 @@ oms_status_enu_t oms3::BusConnector::addConnector(const oms3::ComRef &cref)
...
 
+oms_status_enu_t oms3::BusConnector::deleteConnector(const oms3::ComRef &cref)
+{
+  for (auto it = conrefs.begin(); it != conrefs.end(); ) {
+    if (*it == cref) {
+      conrefs.erase(it);
+      updateConnectors();
+      return oms_status_ok;
+    } else {
+      ++it;
```
Comment (lochel) [inline src/OMSimulatorLib/BusConnector.cpp]: Why not having the increment in the loop header?

**Comment by lochel** [comment]
I guess you have to rebase the changes manually, because Hudson will probably fail on it.

**Comment by adeas31** [comment]
Fixed it.

**Comment by robbr48** [comment]
Wouldn't "oms3_removeConnectorFromBus" be a better name?

**Comment by adeas31** [comment]
> Wouldn't "oms3_removeConnectorFromBus" be a better name?

Yeah I thought about it but then I saw we have used `delete` prefix earlier e.g., `deleteSubModel`, `deleteConnection` etc. I thought it makes sense to have some symmetry in the API naming.

**Comment by OpenModelica-Hudson** [comment]
The following pull requests are now being [tested](https://test.openmodelica.org/hudson/job/OMSimulator-PR/703//console). It may take a while, depending on the workload.
  - OpenModelica/OMSimulator#408

The changes will be pushed if all tests run successfully.

**Comment by OpenModelica-Hudson** [comment]
All tests were successful. :sun_with_face:

**Pull request closed without merging**